conjure-undertow-annotations/src/main/java/com/palantir/conjure/java/undertow/annotations/DefaultSerDe.java

```diff
@@ -40,22 +41,22 @@
 
     @Override
     @SuppressWarnings("unchecked")
-    public <T> Deserializer<T> deserializer(TypeMarker<T> marker, UndertowRuntime runtime) {
+    public <T> Deserializer<T> deserializer(TypeMarker<T> marker, UndertowRuntime runtime, Endpoint endpoint) {
         if (INPUT_STREAM.getType().equals(marker.getType())) {
             return (Deserializer<T>) new BinaryDeserializer(runtime.bodySerDe());
         }
-        return runtime.bodySerDe().deserializer(marker);
+        return runtime.bodySerDe().deserializer(marker, endpoint);
     }
 
     @Override
     @SuppressWarnings("unchecked")
-    public <T> Serializer<T> serializer(TypeMarker<T> marker, UndertowRuntime runtime) {
+    public <T> Serializer<T> serializer(TypeMarker<T> marker, UndertowRuntime runtime, Endpoint endpoint) {
         Type type = marker.getType();
         Serializer<T> maybeSpecialSerializer = (Serializer<T>) maybeSpecialSerializer(type, runtime);
```

Comment:
I found it a bit confusing, that for some serde types, we want to provide the endpoints metadata while for others (e.g. `BinaryDeserializer`) we don't (similar to https://github.com/palantir/conjure-java/pull/1650#discussion_r806137596).

Can you elaborate a bit on this, Carter? Using the example of the `request-size-limit feature`, is this generally not wanted for streaming input/output? I could imagine a use-case where we want to wrap the input stream and restrict it's total size.

Reply:
Of course! There's a distinction between binary and structured deserializers. Binary data is not serialized, and doesn't need to be deserialized, the data is passed directly to the consumer. Secondly, binary data is streamed to the service, where the consumer is responsible for handling the data -- there's no reason a multi-gigabyte request couldn't be handled gracefully in a streaming fashion without risking oom, however if the implementation chooses to buffer, it can also length-check (in contrast to the structured data deserializers which fully abstract away the stream access).

Hopefully that makes some kind of sense